### PR TITLE
make sure static analysis realizes err_sys does exit()

### PR DESCRIPTION
**Pull request by toddouska**

### CODE REVIEW

**Comment by ejohnstown** [comment]
I tried applying this in wolfSSH and clang complained that the err_sys() function should (can) have the noreturn attribute. I also just tried wolfSSL with this change, and get the same error. (Apple LLVM version 7.3.0 (clang-703.0.31))

**Comment by kaleb-himes** [comment]
clang will still throw a warning here: complains that function can be declared as:

```
static INLINE __attribute__((noreturn)) void err_sys(const char* msg)
```

**Comment by kaleb-himes** [comment]
Still trying to determine why the introduction of the macros:

```
AssertIntEq...
AssertFalse...
```

caused this report to be generated.

File: wolfssl/test.h [hunk 1]
```diff
@@ -355,11 +355,29 @@ void join_thread(THREAD_TYPE);
...
 static const word16      wolfSSLPort = 11111;
 
-static INLINE void err_sys(const char* msg)
+
+#if defined(__GNUC__)
+    #define WC_NORETURN __attribute__((noreturn))
+#else
+    #define WC_NORETURN
+#endif
+
```
Comment (dgarske) [inline wolfssl/test.h]: Perhaps WC_NORETURN would be more "at home" in types.h?

Comment (toddouska) [inline wolfssl/test.h]: I considered that but... I thought we shouldn't consider no return hacks in the library proper.  Library functions should return, we should leave exit() and abort() to the user.  For testing though, where we ARE the user, it makes more sense for this hack.

Comment (dgarske) [inline wolfssl/test.h]: Thanks for explaining your thoughts behind it. Makes sense and I agree. We don't want to encourage WC_NORETURN use elsewhere let's keep it here.

**Comment by dgarske** [comment]
Thanks Todd for this fix. This solution ensures compatibility with other compilers. Thanks for documenting the "if(msg)" in err_sys.

**Pull request merged**